client/src/pages/StadiumProfile.tsx

```diff
@@ -1,10 +1,51 @@
 import { Box } from '@mui/system';
-import ImageSlider from '../components/stadiumProfile/ImageSlider';
+import { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import axios from 'axios';
+import {
+  BioSection,
+  ImageSlider,
+  ProfileCalender,
+} from '../components/stadiumProfile';
+import { UserData } from '../interfaces/StadiumProfile';
 
 const StadiumProfile = () => {
+  const [userData, setUserData] = useState<UserData | null>(null);
+  const [gallery, setGallery] = useState({
+    id: 0,
+    image1: '',
+    image2: '',
+    image3: '',
+    image4: '',
+  });
+
+  const { id } = useParams();
+  const fetchProfileData = async (stadiumId: string) => {
+    const { data } = await axios.get(`/api/v1/stadiums/profile/${stadiumId}`);
+
+    setGallery(data.data.Stadium.stadiumGallery[0]);
+    setUserData(data.data as UserData);
+  };
+
+  useEffect(() => {
```

Comment:
include id in the dependency array 
 useEffect(() => {
    if (id) {
      fetchProfileData(id);
    }
  }, [id]);